src/wasm/wasm-binary.cpp

```diff
@@ -731,7 +732,42 @@ void WasmBinaryWriter::finishUp() {
 
 // reader
 
+bool WasmBinaryBuilder::hasDWARFSections() {
+  assert(pos == 0);
+  getInt32(); // magic
+  getInt32(); // version
+  bool has = false;
+  while (more()) {
```

Comment:
It seems kind of crazy that we don't already have some kind of general section-iteration mechanism. But I guess that's not the kind of thing Binaryen is usually used for...
This is ok for now (assuming it doesn't duplicate something we do have elsewehere) but let's try not to write this code again in the future.